svm/src/transaction_processor.rs

```diff
@@ -494,45 +488,89 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         let fee_payer_index = 0;
         validate_fee_payer(
             fee_payer_address,
-            &mut fee_payer_account,
+            &mut loaded_fee_payer.account,
             fee_payer_index,
             error_counters,
             rent_collector,
             fee_details.total_fee(),
         )?;
 
+        if let Some(ref advanced_nonce_info) = advanced_nonce {
+            Self::validate_transaction_nonce(
+                account_loader,
+                advanced_nonce_info,
+                durable_nonce,
+                error_counters,
+            )?;
+        }
+
         // Capture fee-subtracted fee payer account and next nonce account state
         // to commit if transaction execution fails.
         let rollback_accounts = RollbackAccounts::new(
-            nonce,
+            advanced_nonce,
             *fee_payer_address,
-            fee_payer_account.clone(),
-            fee_payer_rent_debit,
+            loaded_fee_payer.account.clone(),
+            loaded_fee_payer.rent_collected,
             fee_payer_loaded_rent_epoch,
         );
 
         Ok(ValidatedTransactionDetails {
             fee_details,
             rollback_accounts,
             compute_budget_limits,
-            loaded_fee_payer_account: LoadedTransactionAccount {
-                loaded_size: fee_payer_account.data().len(),
-                account: fee_payer_account,
-                rent_collected: fee_payer_rent_debit,
-            },
+            loaded_fee_payer_account: loaded_fee_payer,
         })
     }
 
+    fn validate_transaction_nonce<CB: TransactionProcessingCallback>(
```

Comment:
right, and we also have to cover cases where it is reopened, where it is advanced by a blockhash transaction, etc

the only way we can have batched nonce validation and also achieve the goal that batching has no effect on consensus is to change the nonce instructions to ban cpi. then we could easily verify all nonce uses are valid by looking at the instructions called and the initial account states

Reply:
actually nevermind, its impossible. you must execute the transactions first because a blockhash transaction could attempt to advance, authorize, or close a nonce account, but then fail for unrelated reasons and leave the state unchanged